l3_footstep_planning_plugins/include/l3_footstep_planning_plugins/base/hlut_heuristic_plugin.h

```diff
@@ -0,0 +1,190 @@
+// ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//=================================================================================================
+
+#ifndef L3_FOOTSTEP_PLANNING_PLUGINS_HLUT_HEURISTIC_PLUGIN_H
+#define L3_FOOTSTEP_PLANNING_PLUGINS_HLUT_HEURISTIC_PLUGIN_H
+
+#include <queue>
+
+#include <grid_map_ros/grid_map_ros.hpp>
+
+#include <l3_libs/conversions/l3_msg_conversions.h>
+#include <l3_libs/robot_description/base_info.h>
+
+#include <l3_footstep_planning_libs/types/heuristic_lookup_table.h>
+
+#include <l3_footstep_planning_plugins/aggregator/world_model.h>
+#include <l3_footstep_planning_plugins/base/heuristic_plugin.h>
+
+namespace std
+{
+template <>
+struct hash<l3::PositionIndex>
+{
+  size_t operator()(const l3::PositionIndex& index) const
+  {
+    // cantor pairing function
+    return ((index.x() + index.y()) * (index.x() + index.y() + 1) >> 1) + index.y();
+  }
+};
+}  // namespace std
+
+namespace l3_footstep_planning
+{
+/**
+ * @brief Base class for heuristic plugins that use a heuristic lookup table.
+ *
+ * @param resolution The resolution of the HLUT.
+ * @param size_x The size of the HLUT in x-direction.
+ * @param size_y The size of the HLUT in y-direction.
+ * @param check_foothold_accessibility Specifies whether the foothold of the robot should be used for the accessibility check.
+ * @param check_floating_base_accessibility Specifies whether the floating base of the robot should be used for the accessibility check.
+ * @param visualize Specifies whether the HLUT should be visualized.
+ * @param vis_frame_id The frame id of the HLUT visualization.
+ * @param vis_topic The topic of the HLUT visualization.
+ * @param vis_layer The layer of the HLUT visualization.
+ */
+class HLUTHeuristicPlugin : public HeuristicPlugin
+{
+public:
+  /**
+   * @brief Default constructor.
+   */
+  HLUTHeuristicPlugin(const std::string& name);
+
+  bool loadParams(const vigir_generic_params::ParameterSet& params) override;
+
+  bool initialize(const vigir_generic_params::ParameterSet& params) override;
+
+  void preparePlanning(const msgs::StepPlanRequest& req) override;
+
+  double getHeuristicValue(const State& from, const State& to, const State& start, const State& goal) const override;
+
+  double getHeuristicValue(const Foothold& from, const Foothold& to, const State& start, const State& goal) const override;
+
+  double getHeuristicValue(const FloatingBase& from, const FloatingBase& to, const State& start, const State& goal) const override;
+
+protected:
+  /**
+   * @brief An entry of HLUT consisting of its index and heuristic value.
+   */
+  struct hlutEntry
+  {
+    // The index of the entry.
+    l3::PositionIndex index;
+
+    // The heuristic value of the entry.
+    float heuristic_distance;
```

Comment:
Comment says "value" but member is named "distance". Should be renamed to "value",